files/scripts/syncd.sh

```diff
@@ -154,7 +189,12 @@ stop() {
         /bin/systemctl stop pmon
         debug "Stopped pmon service"
     fi
-
+    if [[ x$sonic_asic_platform == x"mellanox" ]]; then
+        # In case of SAI failure the last line of sairedis.rec would contain switch_shutdown_request
+        if tail -1 /var/log/swss/sairedis.rec | grep -q switch_shutdown_request; then
+            collect_mst
```

Comment:
>collect_mst [](http://example.com/codeflow?start=12&length=11)

Should this PR be raised on master first, and they backport to all the branches (>= 201911)?

Reply:
Still applicable.

Reply:
@qiluo-msft This change is not required for master. In master we have an event driven firmware tool which will automatically create these dumps on any firmware failure event.

Reply:
How about branches between (201911, master)

Reply:
It exists from 202012 and so we are good.